caput/interferometry.py

```diff
@@ -0,0 +1,121 @@
+"""Useful functions for radio interferometry.
+
+Coordinates
+===========
+
+.. autosummary::
+    :toctree: generated/
+
+    sph_to_ground
+    project_distance
+
+
+Interferometry
+==============
+
+.. autosummary::
+    :toctree: generated/
+
+    fringestop_phase
+"""
+# === Start Python 2/3 compatibility
+from __future__ import absolute_import, division, print_function, unicode_literals
+from future.builtins import *  # noqa  pylint: disable=W0401, W0614
+from future.builtins.disabled import *  # noqa  pylint: disable=W0401, W0614
+
+# === End Python 2/3 compatibility
+
+import numpy as np
+
+
+def sph_to_ground(ha, lat, dec):
```

Comment:
I verified that is equivalent to my routine to within round-off error.  My version uses a rotation matrix, but I like this better since it has fewer lines and better supports broadcasting if `ha` and `dec` have different shapes.

Reply:
I also have the inverse function if that would be useful (I can also make my own pull request):

```
def ground_to_sph(x, y, lat):
    """Get the CIRS coordinates.
    Latitude is given in radians. Assumes z is positive
    Parameter
    ---------
    x : array_like
        The East projection of the angular position
    y : array_like
        The North projection of the angular position
    lat : array_like
        The latitude of the observatory.
    Returns
    -------
    HA, dec: array_like
        Hour Angle and declination in radians
    """

    z = np.sqrt(1 - x**2 - y**2)
    
    xe = z * np.cos(lat) - y * np.sin(lat)
    ye = x
    ze = y * np.cos(lat) + z * np.sin(lat)
    
    HA = -1 * np.arctan2(ye, xe)
    dec = np.arctan2(ze, np.sqrt(xe**2 + ye**2))
    
    return HA, dec
```